match_sorts_to_ref.py

```diff
@@ -0,0 +1,48 @@
+import os
+import sys
+import argparse
+from dateutil.parser import *
+import datetime
+
+import pandas as pd
+import numpy as np
+
+def match_sorts(ARGS):
+    '''
+    match extracted RS numbers and dates to reference sort data
+    '''
+    extracted = pd.read_csv('{}{}{}'.format(ARGS.dir, os.path.sep, ARGS.data))
+    ref = pd.read_excel('{}{}{}'.format(ARGS.dir, os.path.sep, ARGS.ref))
+
+    def get_ret(row):
+        # match sort number IFF the date and at least one of the RS numbers matches
+        rs_set = row['RAW_RS_NUM'].upper().split(';') + [row['PROCESSED_RS_NUM'].upper()]
+        print ('{} - {}'.format(row['SortDate'], rs_set))
+        ret = ref.loc[(ref['SortDate'] == row['SortDate']) & \
+            ((ref['RSNumDNA'].isin(rs_set)) | (ref['RSNumKi'].isin(rs_set)))]
+        if len(set(ret['SortNumSD'])) == 1:
+            
+            return ret  
+    
+    extracted.apply(get_ret, axis=1)   
+    return extracted
+
+
+def parse_arguments(parser):
+    parser.add_argument('--dir', type=str, help='base directory',
+            default='C:/Users/esilgard/Fred Hutchinson Cancer Research Center/Sanchez, Carissa A - SortScanning')
```

Comment:
do you want to point it to data-science folder?

Reply:
no -the data doesn't live in the data science folder, it's in the Reid lab One Drive - mapped to my C atm